doc/man3/BIO_read.pod

```diff
@@ -36,6 +36,8 @@ however; for example, BIO_gets() on a digest BIO will calculate and
 return the digest and other BIOs may not support BIO_gets() at all.
 The returned string is always NUL-terminated and the '\n' is preserved
 if present in the input data.
+On binary input there may be NUL characters within the string;
```

Comment:
I would be very hesitant changing the semantics of BIO_gets() like this. It could have serious unwanted consequences on third party code.

Reply:
All right, reverted the changes to `BIO_gets()`. Instead, documented its bug on input containing the NUL char
and added a new independent function `BIO_get_line()` that works correctly on binary input (which include NUL).
This function is now used where needed in `asn_mime.c`: in `multi_split().`